packages/calypso-e2e/src/lib/components/editor-window.ts

```diff
@@ -0,0 +1,60 @@
+import { Page, Frame } from 'playwright';
+
+/** */
+export class EditorWindow {
+	public page: Page;
+
+	/** */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/** */
+	async getEditorFrame(): Promise< Page | Frame > {
```

Comment:
I have a couple of questions here.

1. Was it difficult or unfeasible to return a single value in order to abstract the underlying implementation?
2. What's the reason for returning a `Page | Frame` instead of the more "modern" `Locator | FrameLocator`?

Let me elaborate for each question.

For 1, in the current implementation, I tried to present a unified interface in the form of returning a `Locator` to a top-level element of the page (un-iframed) or the first feasible element (iframed). You can see this in [these lines](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/pages/editor-page.ts#L87-L90).

While the interfaces of Frame and Page are nearly identical, I would argue that it would be preferable to have a single return value so the consumers aren't having to cast the types.


Point 2 is the one that I comparatively have a stronger opinion on. 
I feel that we should use Locators as much as possible because Frames and many Page methods are being de-emphasized lately. I'm wondering if you chose to use Page/Frame because Locators haven't yet implemented a `waitForFunction` method yet.

Reply:
> The implementation in this PR allows for an easy transition from Page/Frame to Locator, so I'll test that on another branch, and we'll decide which one would be best to go with. How does that sound?

That sounds fair to me. Let's keep it simple(r) in this PR by sticking to the Frame/Page dual-type or a single Frame return type.